front/src/modules/trainschedule/components/Timetable/TimetableToolbar.tsx

```diff
@@ -205,41 +247,79 @@ const TimetableToolbar = ({
     return `${pacedTrainLabel}\u00A0${t('translation:common.and')}\u00A0${trainScheduleLabel}`;
   };
 
+  // TODO : doesn't work for german. Find a better way to handle this
+  const deleteTimetableItemsComputedLabel = () => {
+    const confirmLabel = t('common/common:actions.deleteSelectionCount');
+
+    // We don't want to display the selected ratio here
+    const pacedTrainLabel = t('pacedTrainCount', {
+      count: 0,
+      totalCount: selectedPacedTrainIds.length,
+    });
+
+    const trainScheduleLabel = t('trainCount', {
+      count: 0,
+      totalCount: selectedTrainScheduleIds.length,
+    });
+
+    let itemsLabel = '';
+
+    if (
+      totalPacedTrainCount === 0 ||
+      (selectedPacedTrainIds.length > 0 && selectedTrainScheduleIds.length === 0)
+    ) {
+      itemsLabel = pacedTrainLabel;
+    }
+
+    if (
+      totalPacedTrainCount === 0 ||
+      (selectedTrainScheduleIds.length > 0 && selectedPacedTrainIds.length === 0)
+    ) {
+      itemsLabel = trainScheduleLabel;
+    }
+
+    if (selectedPacedTrainIds.length > 0 && selectedTrainScheduleIds.length > 0) {
+      itemsLabel = `${pacedTrainLabel}\u00A0${t('translation:common.and')}\u00A0${trainScheduleLabel}`;
+    }
```

Comment:
You don't need to test the total counts here, only the selected counts matter. You can simplify the code like this:
```suggestion
    if (selectedPacedTrainIds.length > 0 && selectedTrainScheduleIds.length > 0) {
      itemsLabel = `${pacedTrainLabel}\u00A0${t('translation:common.and')}\u00A0${trainScheduleLabel}`;
    } else {
      itemsLabel = selectedPacedTrainIds.length > 0 ? pacedTrainLabel : trainScheduleLabel;
    }

```

Reply:
After a comment from @emersion I changed this function a bit. Let me know if it's ok for you.

Reply:
Yes perfect ! I hesitated to ask the same thing :star_struck: